cookbook/integrations/kubernetes/k8s_spark/README.rst

```diff
@@ -5,20 +5,64 @@ Spark on Kubernetes
 
 Flyte can execute Spark jobs natively on a Kubernetes Cluster, which manages the lifecycle, spin-up, and tear down
 of a virtual cluster. It leverages the open-sourced `Spark On K8s Operator <https://github.com/GoogleCloudPlatform/spark-on-k8s-operator>`_
-and can be enabled without signing up for any service. 
+and can be enabled without signing up for any service. This is like running an ``transient cluster`` which are better for production workloads, but have an extra cost of setup and teardown.
+In Flyte/K8s, the cost is ammortized by the fact that pods are faster to create than a machine, but the penalty of downloading docker images may affect the performance. Also remember - starting a pod is not as fast as running a process.
+
 
 Flytekit makes it possible to write pyspark code natively as a task and the
 Spark cluster will be automatically configured using the decorated ``SparkConf``. The examples in this section provide a
 hands-on tutorial for writing pyspark tasks.
 
-Environment Setup
-------------------
+.. NOTE::
+
+   This has been tested at scale and more than 100k Spark Jobs run through Flyte at Lyft. This still needs a large capacity on Kubernetes and careful configuration.
+   We recommend using Multi-cluster mode - :ref:`howto-multi-cluster`, and enabling :ref:`howto-resource-quota` for large and extremely frequent Spark Jobs.
+   For extremely short running jobs, this is still not a recommended approach, and it might be better to use a pre-spawned cluster.
```

Comment:
nit: Let's add some number ? maybe say _(< 2-3 mins)_ jobs for extremely short running jobs .

Reply:
great, thank you!